### ISSUE TITLE
梯度裁剪相关问题

### ISSUE THREAD

**Issue by Celina-love-sweet** [modelscope/ms-swift#4504]
您好，我能不能控制让每步的梯度都不超过1呢，有个参数max_grad_norm设置了1没有用，还是输出这么多，这样会导致模型发散：{'loss': 2.80872822, 'seq_acc': 0.0, 'grad_norm': 84.40345001, 'learning_rate': 5e-07, 'memory(GiB)': 67.7, 'train_speed(iter/s)': 0.045443, 'epoch': 0.0, 'global_step/max_steps': '1/1585', 'percentage': '0.06%', 'elapsed_time': '21s', 'remaining_time': '9h 30m 4s'}

**Comment by Celina-love-sweet** [comment]
> 您好，我能不能控制让每步的梯度都不超过1呢，有个参数max_grad_norm设置了1没有用，还是输出这么多，这样会导致模型发散：{'loss': 2.80872822, 'seq_acc': 0.0, 'grad_norm': 84.40345001, 'learning_rate': 5e-07, 'memory(GiB)': 67.7, 'train_speed(iter/s)': 0.045443, 'epoch': 0.0, 'global_step/max_steps': '1/1585', 'percentage': '0.06%', 'elapsed_time': '21s', 'remaining_time': '9h 30m 4s'}

我设置了80也没有用，这个梯度太大了，我想要让梯度值一直最多不会超过1.

**Comment by Celina-love-sweet** [comment]
> > 您好，我能不能控制让每步的梯度都不超过1呢，有个参数max_grad_norm设置了1没有用，还是输出这么多，这样会导致模型发散：{'loss': 2.80872822, 'seq_acc': 0.0, 'grad_norm': 84.40345001, 'learning_rate': 5e-07, 'memory(GiB)': 67.7, 'train_speed(iter/s)': 0.045443, 'epoch': 0.0, 'global_step/max_steps': '1/1585', 'percentage': '0.06%', 'elapsed_time': '21s', 'remaining_time': '9h 30m 4s'}
> 
> 我设置了80也没有用，这个梯度太大了，我想要让梯度值一直最多不会超过1.

swift的版本是3.2.0

**Comment by Celina-love-sweet** [comment]
> 您好，我能不能控制让每步的梯度都不超过1呢，有个参数max_grad_norm设置了1没有用，还是输出这么多，这样会导致模型发散：{'loss': 2.80872822, 'seq_acc': 0.0, 'grad_norm': 84.40345001, 'learning_rate': 5e-07, 'memory(GiB)': 67.7, 'train_speed(iter/s)': 0.045443, 'epoch': 0.0, 'global_step/max_steps': '1/1585', 'percentage': '0.06%', 'elapsed_time': '21s', 'remaining_time': '9h 30m 4s'}
> 
> 这个是训练参数 CUDA_VISIBLE_DEVICES=0,1,2,3 swift sft --model /data1_hdd/junxi/InternVL2.5/OpenGVLab/InternVL2_5-8B --dataset /data1_hdd/junxi/InternVL2.5/mydata/fakesv_train.jsonl --val_dataset /data1_hdd/junxi/InternVL2.5/mydata/fakesv_val.jsonl /data1_hdd/junxi/InternVL2.5/mydata/fakesv_test.jsonl --acc_strategy seq --system fakesv.txt --device_map mapmoe.json --train_type lora --torch_dtype bfloat16 --attn_impl flash_attn --num_train_epochs 5 --per_device_train_batch_size 4 --split_dataset_ratio 0.0 --per_device_eval_batch_size 1 --max_grad_norm 1 --learning_rate 8e-5 --lora_rank 8 --lora_alpha 32 --target_modules all-linear --freeze_vit true --freeze_aligner true --gradient_accumulation_steps 2 --save_strategy epoch --eval_strategy epoch --eval_steps 1 --save_steps 1 --save_total_limit 5 --logging_steps 10 --max_length 8192 --modules_to_save model.switchTransformer_coarse_grained model.switchTransformer_fine_grained classifier attn_score --output_dir your_path --warmup_ratio 0.1 --data_seed 2025 --seed 2025 --temperature 0 --dataloader_num_workers 4 \

我尝试了单独使用原始的internvl2.5模型，那个梯度裁剪也没有效果，swift版本是3.2.0，我的理解是--max_grad_norm 1设置后，梯度最大不会超过1的。

**Comment by Celina-love-sweet** [comment]
@Jintao-Huang @hjh0119 @tastelikefeet

**Comment by Celina-love-sweet** [comment]
> [@Jintao-Huang](https://github.com/Jintao-Huang) [@hjh0119](https://github.com/hjh0119) [@tastelikefeet](https://github.com/tastelikefeet)

球球大佬们救救
